torch/_torch_docs.py

```diff
@@ -449,6 +449,14 @@ def parse_kwargs(desc):
 will be performed.  Similarly, if the data is an ``ndarray`` of the corresponding `dtype` and the `device` is the cpu,
 no copy will be performed.
 
+.. warning::
+
+    When attr:`data` is a tensor `x` with different attr:`dtype` or attr:`device`,
+    :func:`as_tensor()` reads out 'the data' from whatever it is passed,
+    and constructs a leaf variable. Therefore ``tensor.as_tensor(x, dtype=dtype, device=device)``
+    is equivalent to ``x.clone().detach().to(dtype, device)``.
+    The equivalents using ``clone()`` and ``detach()`` are recommended.
```

Comment:
Hmmm I'm not sure if that's a good decision. I'd expect `torch.as_tensor(tensor)` to return `tensor` unmodified (and retaining its autograd history).

Reply:
I also don't think it matches what @gchanan said above. There's no `clone()` happening in `as_tensor`.

Reply:
Yes, agreed upon the point on `torch.as_tensor(tensor)`! Sorry I haven't push the update to address @gchanan comments. Also, for `torch.as_tensor(tensor, dtype=dtype, device=device)`, what should the copy constructed tensor look like? Should it be made non-differentiable?